grudge/op.py

```diff
@@ -263,19 +270,21 @@ def _div_helper(dcoll: DiscretizationCollection, diff_func, vecs):
         return result
 
 
-def local_div(dcoll: DiscretizationCollection, vecs):
+def local_div(dcoll: DiscretizationCollection, vecs) -> ArrayOrContainerT:
     r"""Return the element-local divergence of the vector function
     :math:`\mathbf{f}` represented by *vecs*:
 
     .. math::
 
         \nabla|_E \cdot \mathbf{f} = \sum_{i=1}^d \partial_{x_i}|_E \mathbf{f}_i
 
-    :arg vec: an object array of
-        a :class:`~meshmode.dof_array.DOFArray`\ s,
+    :arg vecs: an object array of
+        :class:`~meshmode.dof_array.DOFArray`\ s or
+        :class:`~arraycontext.container.ArrayContainer` objects,
         where the last axis of the array must have length
         matching the volume dimension.
-    :returns: a :class:`~meshmode.dof_array.DOFArray`.
+    :returns: a :class:`~meshmode.dof_array.DOFArray` or an
+        :class:`~arraycontext.container.ArrayContainer` of them.
```

Comment:
I don't think `_div_helper` deals with anything other than object arrays.

Reply:
I believe this is all cleared up in https://github.com/inducer/grudge/pull/154/commits/a87734006d247785f3c5baf43bde860c46dbd0c9. `_div_helper` is intentionally only working on object arrays, as the container is expected to have vector (obj array) entries.